hydra_agent/agent.py

```diff
@@ -10,26 +12,39 @@
 logger = logging.getLogger(__file__)
 
 
-class Agent(Session):
+class Agent(Session, socketio.ClientNamespace, socketio.Client):
     """Provides a straightforward GET, PUT, POST, DELETE -
     CRUD interface - to query hydrus
     """
-    def __init__(self, entrypoint_url: str) -> None:
+
+    def __init__(self, entrypoint_url: str, namespace: str='/sync') -> None:
         """Initialize the Agent
         :param entrypoint_url: Entrypoint URL for the hydrus server
+        :param namespace: Namespace endpoint to listen for updates
         :return: None
         """
         self.entrypoint_url = entrypoint_url.strip().rstrip('/')
         self.redis_proxy = RedisProxy()
         self.redis_connection = self.redis_proxy.get_connection()
-        super().__init__()
+        Session.__init__(self)
         jsonld_api_doc = super().get(self.entrypoint_url + '/vocab').json()
         self.api_doc = doc_maker.create_doc(jsonld_api_doc)
         self.initialize_graph()
-        self.graph_operations = GraphOperations(entrypoint_url,
+        self.graph_operations = GraphOperations(self.entrypoint_url,
                                                 self.api_doc,
                                                 self.redis_proxy)
 
+        # Declaring Socket Rules and instaciating Synchronization Socket
+        socketio.ClientNamespace.__init__(self, namespace)
+        socketio.Client.__init__(self, logger=True)
+        socketio.Client.register_namespace(self, self)
+        # After hydrus implementation connect directly to the sv, self.entrypoint
+        self.entrypoint_url_temp = 'http://localhost:5000'
```

Comment:
Don't you think this shouldn't be hardcoded? Rest everything is looking good. 👍

Reply:
Yeah, that was weak memory, sorry. I fixed all connections to fetch the same link as the hydrus server, so it's working with Vishal's implementation now.